flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PartitionData.java

```diff
@@ -45,6 +51,14 @@
 
 	public abstract boolean isBuffer();
 
+	/**
+	 * Returns the buffer-format partition data with the provided memory segment or not.
+	 *
+	 * @param segment it might be needed to read the partition data into.
+	 * @return the buffer represents the partition data.
+	 */
+	public abstract Buffer getBuffer(@Nullable MemorySegment segment) throws IOException;
```

Comment:
I would change this to `Supplier<MemorySegment>` so that the code lazily fetches (and allocates) the memory segment only when needed. See also next comment about lazy creation.